File: library/src/main/java/org/meganekkovr/animation/RotationUpdateListener.java
package org.meganekkovr.animation;

import android.animation.ValueAnimator;
import android.support.annotation.NonNull;

import org.joml.Quaternionf;
import org.meganekkovr.Entity;

class RotationUpdateListener extends TransformUpdateListener {

    public RotationUpdateListener(@NonNull Entity target) {
        super(target);
    }

    @Override
    public void onAnimationUpdate(@NonNull ValueAnimator animation) {
        target.setRotation((Quaternionf) animation.getAnimatedValue());
    }
}

File: library/src/main/java/org/meganekkovr/animation/RotationUpdateListener.kt
package org.meganekkovr.animation

import android.animation.ValueAnimator

import org.joml.Quaternionf
import org.meganekkovr.Entity

internal class RotationUpdateListener(target: Entity) : TransformUpdateListener(target) {

    override fun onAnimationUpdate(animation: ValueAnimator) {
        target.rotation = animation.animatedValue as Quaternionf
    }
}
